source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -60,6 +60,15 @@ void SslSocket::setTransportSocketCallbacks(Network::TransportSocketCallbacks& c
   ASSERT(!callbacks_);
   callbacks_ = &callbacks;
 
+  provider_ = ctx_->getPrivateKeyOperationsProvider();
+  if (provider_) {
+    ops_ = provider_->getPrivateKeyOperations(*this, callbacks_->connection().dispatcher());
+    Ssl::PrivateKeyMethodSharedPtr private_key_methods = ops_->getPrivateKeyMethods(ssl_.get());
+    if (private_key_methods) {
+      SSL_set_private_key_method(ssl_.get(), private_key_methods.get());
```

Comment:
I guess this function should be incapsulated somewhere inside `source/extensions/transport_socket_tls/private_key` to avoid messing with OpenSSL's API if we add conditional OpenSSL builds.

Reply:
You're right -- the setup could also be in a `PrivateKeyImpl` class or similar. On the other hand the whole concept of BoringSSL private key calls doesn't make sense if OpenSSL is used.